pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
-                devices = list(range(self.tpu_cores))
-        elif self.use_ipu:
-            devices = list(range(self.num_ipus))
+    def _choose_strategy(self) -> None:
+        if self._accelerator_flag == "ipu":
+            self._strategy_flag = "ipu_strategy"
+        elif self._accelerator_flag == "tpu":
+            if self._parallel_devices and len(self._parallel_devices) > 1:
+                self._strategy_flag = "tpu_spawn"
+            else:
+                # TODO: lazy initialized device, then here could be self._strategy_flag = "single_tpu_device"
+                self._strategy_flag = SingleTPUStrategy(device=self._parallel_devices[0])
+        elif _HOROVOD_AVAILABLE and ("OMPI_COMM_WORLD_RANK" in os.environ or "HOROVOD_RANK" in os.environ):
+            self._strategy_flag = "horovod"
         else:
-            devices = [torch.device("cpu")] * self.num_processes
-        return devices
+            if self._num_nodes_flag > 1:
+                self._strategy_flag = "ddp"
+            elif len(self._parallel_devices) <= 1:
+                device = (
+                    device_parser.determine_root_gpu_device(self._parallel_devices)
+                    if self._accelerator_flag == "gpu"
+                    else "cpu"
+                )
+                # TODO: lazy initialized device, then here could be self._strategy_flag = "single_device"
+                self._strategy_flag = SingleDeviceStrategy(device=device)
+            elif len(self._parallel_devices) > 1:
+                self._strategy_flag = "ddp_spawn"
+            else:
+                self._strategy_flag = "ddp"
 
-    @property
-    def root_gpu(self) -> Optional[int]:
-        return (
-            self.strategy.root_device.index
-            if not isinstance(self.accelerator, (IPUAccelerator, TPUAccelerator))
-            else None
-        )
+    def _check_strategy_and_fallback(self) -> None:
+        """Checks edge cases when the strategy selection was a string input, and we need to fall back to a
+        different choice depending on other parameters or the environment."""
+        # current logic, fallback only apply to user pass in str config not object config
+        strategy_flag = "" if isinstance(self._strategy_flag, Strategy) else self._strategy_flag
 
-    @staticmethod
-    def _is_plugin_training_type(plugin: Union[str, Strategy]) -> bool:
-        if isinstance(plugin, str) and (plugin in StrategyRegistry or plugin in list(_StrategyType)):
-            return True
-        return isinstance(plugin, Strategy)
+        if strategy_flag == "ddp_cpu":
+            if _TPU_AVAILABLE:
+                raise MisconfigurationException(
+                    "`accelerator='ddp_cpu'` is not supported on TPU machines. "
+                    "Learn more: https://github.com/PyTorchLightning/pytorch-lightning/issues/7810"
+                )
+            if self._device_flag == 1 and self._num_nodes_flag > 1:
```

Comment:
Now, DDP is the default with 1 GPU multi node ?

Reply:
Won't it be better to fall back to ddp_spawn for all ?

Reply:
Let's discuss the fallback logic as a follow up in #11449 . This PR is to reproduce existing logic frist, then we can do the follow up items parallel